### ISSUE TITLE
Numbers are indiscriminately turned into links

### ISSUE THREAD

**Issue by awdsns** [dessalines/jerboa#635]
**Jerboa Version**
0.0.34 from Play Store

**Describe the bug**
Any number longer than a single digit in a post or comment text is interpreted as a telephone number and turned into a clickable link.

**To Reproduce**
View any post or comment containing numbers in the app, e.g. https://feddit.de/comment/248034

**Comment by noloman** [comment]
Hi! 
Does this happen in the preview of a comment, when you're in the post list? or does this happen when you're in the post detail?

I noticed that in the post preview (the comment you see when you're in the post list) doesn't render markdown and I'll open a PR to address it.

**Comment by awdsns** [comment]
No, this is not related to preview, just to displaying any text that goes through the markdown renderer. This is what just normal numbers (not formatted as links in the markdown *source*) are displayed:

![grafik](https://github.com/dessalines/jerboa/assets/136314294/1960e916-0431-48ce-9e26-641b835f3cd1)

Markdown preview when writing a comment would be nice, but is a different issue.

**Comment by twizmwazin** [comment]
I suspect that the offending code is the Markwon Linkify plugin we're using here: https://github.com/dessalines/jerboa/blob/main/app/src/main/java/com/jerboa/ui/components/common/MarkdownHelper.kt#L46

**Comment by MV-GH** [comment]
I think so too,
![image](https://github.com/dessalines/jerboa/assets/67873169/27ce598e-5d9d-4722-9d78-b88780f9a95d)
its probably thinking they are phone numbers. It should only do weblinks

**Comment by awdsns** [comment]
Looks like this plugin is also responsible for another unwanted behavior: turning `user@instance` and `!community@instance` into `mailto:` links.
